python/ray/data/datasource/image_folder_datasource.py

```diff
@@ -61,70 +60,60 @@ class ImageFolderDatasource(BinaryDatasource):
 
         >>> import ray
         >>> from ray.data.preprocessors import OrdinalEncoder
-        >>>
         >>> ds = ray.data.read_datasource(  # doctest: +SKIP
         ...     ImageFolderDatasource(),
-        ...     paths=["/data/imagenet/train"]
+        ...     root="/data/imagenet/train",
+        ...     size=(224, 224)
         ... )
         >>> oe = OrdinalEncoder(columns=["label"])  # doctest: +SKIP
-        >>>
         >>> ds = oe.fit_transform(ds)  # doctest: +SKIP
-        >>>
         >>> sample = ds.take(1)[0]  # doctest: +SKIP
         >>> sample["label"]  # doctest: +SKIP
         71
     """  # noqa: E501
 
     def create_reader(
         self,
-        paths: Union[str, List[str]],
-        filesystem: Optional["pyarrow.fs.FileSystem"] = None,
-        partition_filter: PathPartitionFilter = None,
-        **kwargs,
+        root: str,
+        size: Tuple[int, int],
```

Comment:
Do we want to do any sanity check of the user-provided `size`? What's the largest image size we want to support now?

Reply:
Added a check that the size contains positive numbers.

No idea what the large image size should be, if any.